parseddocument.cpp

```diff
@@ -79,36 +79,31 @@ void ParsedDocument::populateReferencesFromModule(const std::unique_ptr<Data::De
     Data::Definitions::Types::const_iterator typeIt;
     for (typeIt = types.begin(); typeIt != types.end(); typeIt++) {
         auto reference = typeIt->second.reference();
-        if (reference.type() == Data::TypeReference::DataType::UserDefined)
+        if (reference.isUserDefined())
             m_refernceLookup.insert(typeIt->second.location().line(), reference);
     }
 }
 
 ModelTreeNode::ModelTreeNodePtr
 ParsedDocument::createDefinition(const std::unique_ptr<Data::Definitions> &definition) const
 {
-    auto definitionNode =
-            ModelTreeNode::ModelTreeNodePtr(new ModelTreeNode(definition->name(),
-                                                              Data::SourceLocation(source().getPath(),
-                                                                                   definition->location().line(),
-                                                                                   definition->location().column())));
+    auto definitionNode = ModelTreeNode::makePtr(definition->name(), Data::Type::UserDefined, updateSourceLocation(definition->location()));
     attachTypesToDefiniton(definition->types(), definitionNode);
 
     return definitionNode;
 }
 
+Data::SourceLocation ParsedDocument::updateSourceLocation(const Data::SourceLocation& location) const
```

Comment:
Would give this function different name, as it does not update the location, but create new one + the amp is in the wrong place.

Reply:
ack, will be `buildLocation` :)